x-pack/legacy/plugins/monitoring/index.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resolve } from 'path';
+import { Server } from 'src/legacy/server/kbn_server';
+import KbnServer from 'src/legacy/server/kbn_server';
+import { LegacyPluginApi, LegacyPluginSpec } from 'src/legacy/plugin_discovery/types';
+import { KIBANA_ALERTING_ENABLED } from './common/constants';
+
+// @ts-ignore
+import { getUiExports } from './ui_exports';
+// @ts-ignore
+import { config as configDefaults } from './config';
+// @ts-ignore
+import { deprecations } from './deprecations';
+// @ts-ignore
+import { Plugin } from './server/plugin';
+// @ts-ignore
+import { initInfraSource } from './server/lib/logs/init_infra_source';
+
+type InfraPlugin = any; // TODO
+type PluginsSetup = any; // TODO
+type LegacySetup = any; // TODO
+
+const deps = ['kibana', 'elasticsearch', 'xpack_main'];
+if (KIBANA_ALERTING_ENABLED) {
+  deps.push(...['alerting', 'actions']);
+}
+
+const validConfigOptions: string[] = [
+  'monitoring.ui.enabled',
+  'monitoring.kibana.collection.enabled',
+  'monitoring.ui.max_bucket_size',
+  'monitoring.ui.min_interval_seconds',
+  'kibana.index',
+  'monitoring.ui.show_license_expiration',
+  'monitoring.ui.container.elasticsearch.enabled',
+  'monitoring.ui.container.logstash.enabled',
+  'monitoring.tests.cloud_detector.enabled',
+  'monitoring.kibana.collection.interval',
+  'monitoring.ui.elasticsearch.hosts',
+  'monitoring.ui.elasticsearch',
+  'monitoring.xpack_api_polling_frequency_millis',
+  'server.uuid',
```

Comment:
you can already start consuming them from `server.newPlatform.setup.core.http.getServerInfo`, server.newPlatform.setup.core.uuid.getInstanceUuid`

Reply:
I think this will also be in the next phase, when we start using np alternatives and whitelisting config properties

Reply:
fair enough